test.sh

```diff
@@ -6,8 +6,37 @@
 Me=$(basename "$0")
 set -euo pipefail
 
-# location of binaries
-BINDIR="${BINDIR:-bin}"
+# location of binaries, which live under $BUILD_DIR, if set.
+build_dir="${BUILD_DIR:-build}"
+BINDIR="${BINDIR:-${build_dir}/bin}"
+
+echo "$Me: build_dir='${build_dir}', BINDIR='${BINDIR}'"
+
+# Location of binaries, which live under $BUILD_DIR, if BINDIR is not set.
+# This means: If BINDIR was -not- set ...
```

Comment:
I don't know what this means, didn't this assignment just happen on 11?

Reply:
I'll fix the comment to read like so: (Would that be more clear?)

```
 14 # If BINDIR -was- set in the user's env, we need to drive off of that.
 15 # To avoid raising a script error by referencing this variable if it was
 16 # -not-set- we just initialized it above to this value. Thus, if the
 17 # following is true, it means the env-var BINDIR was -not- set already.
```

I concede that this script's logic to reliably check if env-var `BINDIR` was set, and not fail, and this comment. ... is cumbersome.

Suggestions to improve this (w/o failing if BINDIR is not set in the env) are welcome!